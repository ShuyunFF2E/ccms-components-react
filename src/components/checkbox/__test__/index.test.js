import React from 'react';
import renderer from 'react-test-renderer';
import { render, mount } from 'enzyme';
import { prefixCls } from '@utils';
import mountTest from '../../../../tests/shared/mountTest';
import Checkbox from '../index';

const classSelector = `${prefixCls}-checkbox`;

describe('Checkbox', () => {
	mountTest(Checkbox);

	const Component = <Checkbox defaultChecked={true} disabled={true} indeterminate={false} value={2} onChange={() => {}} />;

	it('renders correctly', () => {
		const wrapper = render(Component);
		expect(wrapper).toMatchSnapshot();
	});

	it('mount correctly', () => {
		expect(() => renderer.create(Component)).not.toThrow();
	});

	it('change event correctly', () => {
		const onChange = jest.fn();
		const wrapper = mount(<Checkbox onChange={onChange}>Checkbox</Checkbox>);
		wrapper
			.find('input')
			.at(0)
			.simulate('change');
		expect(wrapper.find('label').hasClass(`${classSelector}-checked`)).toBeTruthy();
	});

	it('return value correctly', () => {
		const onChange = jest.fn();
		const wrapper = mount(
			<Checkbox value={2} onChange={onChange}>
				Checkbox
			</Checkbox>
		);
		wrapper
			.find('input')
			.at(0)
			.simulate('change');
		expect(onChange).toHaveBeenCalledWith(true, 2);
	});

	it('disabled changed invalid', () => {
		const onChange = jest.fn();
		const wrapper = mount(
			<Checkbox disabled onChange={onChange}>
				Checkbox
			</Checkbox>
		);
		wrapper
			.find('input')
			.at(0)
			.simulate('change');
		expect(onChange).not.toHaveBeenCalled();
	});
});
