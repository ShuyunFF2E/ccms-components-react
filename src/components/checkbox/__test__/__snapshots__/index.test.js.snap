// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<label
  class="cloud-checkbox cloud-checkbox-disabled cloud-checkbox-checked"
>
  <span
    class="cloud-checkbox-wrapper"
  >
    <input
      class="cloud-checkbox-input"
      disabled=""
      type="checkbox"
      value="2"
    />
    <span
      class="cloud-checkbox-inner"
    />
  </span>
  <span
    class="cloud-checkbox-text"
  />
</label>
`;
